pkg/ddc/alluxio/transform_resources.go

```diff
@@ -60,39 +58,75 @@ func (e *AlluxioEngine) transformResourcesForWorker(runtime *datav1alpha1.Alluxi
 	runtimeInfo, err := e.getRuntimeInfo()
 	if err != nil {
 		e.Log.Error(err, "failed to transformResourcesForWorker")
+		return err
 	}
 	storageMap := tieredstore.GetLevelStorageMap(runtimeInfo)
 
 	e.Log.Info("transformResourcesForWorker", "storageMap", storageMap)
 
-	// TODO(iluoeli): it should be xmx + direct memory
-	memLimit := resource.MustParse("20Gi")
-	if quantity, exists := runtime.Spec.Worker.Resources.Limits[corev1.ResourceMemory]; exists && !quantity.IsZero() {
-		memLimit = quantity
-	}
+
+	// mem set request
+	needUpdated := false
+	var needSetMem resource.Quantity
 
 	for key, requirement := range storageMap {
-		if value.Worker.Resources.Limits == nil {
-			value.Worker.Resources.Limits = make(common.ResourceList)
-		}
 		if key == common.MemoryCacheStore {
 			req := requirement.DeepCopy()
 
-			memLimit.Add(req)
-
-			e.Log.Info("update the requirement for memory", "requirement", memLimit)
+			if runtime.Spec.Worker.Resources.Requests == nil ||
+				runtime.Spec.Worker.Resources.Requests.Memory() == nil ||
+				runtime.Spec.Worker.Resources.Requests.Memory().IsZero() ||
+				req.Cmp(*runtime.Spec.Worker.Resources.Requests.Memory()) > 0 {
+				needUpdated = true
+				needSetMem.Add(req)
```

Comment:
Is req going to be null?

Reply:
no, the req come from: `storageMap := tieredstore.GetLevelStorageMap(runtimeInfo)`

Reply:
> no, the req come from: `storageMap := tieredstore.GetLevelStorageMap(runtimeInfo)`

got it, thank you